src/utils.rs

```diff
@@ -162,37 +162,37 @@ pub(crate) fn random_plusminus_by_size_with_minimum<R: RngCore + CryptoRng>(
 /// [`Transcript`].
 pub(crate) fn plusminus_bn_random_from_transcript(
     transcript: &mut Transcript,
-    n: &BigNumber,
+    _n: &BigNumber,
 ) -> BigNumber {
     let mut is_neg_byte = vec![0u8; 1];
     transcript.challenge_bytes(b"sampling negation bit", is_neg_byte.as_mut_slice());
     let is_neg: bool = is_neg_byte[0] & 1 == 1;
 
     // The sampling method samples from the open interval, so add 1 to sample from
     // the _closed_ interval we want here.
-    let open_interval_max = n + 1;
+    let q = k256_order();
+    let open_interval_max = &q + 1;
     let b = positive_bn_random_from_transcript(transcript, &open_interval_max);
     match is_neg {
         true => -b,
         false => b,
     }
 }
 
-/// Derive a deterministic pseduorandom value in `[0, n)` from the
-/// [`Transcript`].
 pub(crate) fn positive_bn_random_from_transcript(
     transcript: &mut Transcript,
-    n: &BigNumber,
+    _n: &BigNumber,
 ) -> BigNumber {
-    let len = n.to_bytes().len();
+    let len = _n.to_bytes().len();
     let mut t = vec![0u8; len];
```

Comment:
We're using `_n` here to determine the length of the vector that we're sampling from, but then in L295 we're checking if it's less that `q`. I think that `n` is usually way bigger than `q`, so it'll be very rare to sample random bytes that are actually smaller than the bound. If you fix this to be dependent on `q` instead of `_n` it should stop looping forever.

However, in light of the other comment (we _do_ need to take `n`), I think you don't actually need to change this.